### PR TITLE
Mask Generation Tasking (adding about.md)

### PR DESCRIPTION
**Pull request by bellabf**
Hi,

I wrote an "about.md" for the "Mask generation task".
I saw that @NebulaAnish had already opened a PR for the summary, so I paddle back on the "data.ts" in favor of his (thus, the commits).

Thanks!

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
 	"editor.defaultFormatter": "esbenp.prettier-vscode",
 	"editor.formatOnSave": true,
 	"editor.codeActionsOnSave": {
-		"source.fixAll": true
+		"source.fixAll": "explicit"
```
Comment (merveenoyan) [inline .vscode/settings.json:5]: I think you can change this file to the original version so that it will not be included in this PR. 
Hint: since you're only changing files in "packages" folder, doing `git add packages` will prevent adding other unnecessary files for you.

File: packages/tasks/src/tasks/mask-generation/data.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import type { TaskDataCustom } from "..";
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/data.ts:1]: I think it's ok to remove this file if you're not contributing it

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+## Use Cases
+
+### Filtering an Image
+
+The mask might serve as an initial filter to eliminate irrelevant information. For instance, when monitoring vegetation in satellite imaging, a mask can be created to identify green spots, highlighting the relevant region of the image.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:5]:
```suggestion
When filtering for an image, the generated masks might serve as an initial filter to eliminate irrelevant information. For instance, when monitoring vegetation in satellite imaging, mask generation models identify green spots, highlighting the relevant region of the image.
```

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+### Pre-processing
+
+Generating masks that can to separate background and foreground (or other interesting properties) in an image can be done to facilitate learning, especially in semi- or unsupervised learning.
+
+### Human-in-the-loop
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:11]: Can you elaborate the use case here? i.e. The name of the use case felt a little ambiguous

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+## Inference
+
+This section should have useful information about how to pull a model from Hugging Face Hub which is a part of a library specialized in a task and use it.
+
+## Useful Resources
+
+Would you like to learn more about mask generation? Great! Here you can find some curated resources that you may find helpful!
+
+- [Simple mask generation using thresholding in Scikit-learn](https://scikit-image.org/docs/stable/auto_examples/segmentation/plot_thresholding.html)
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:43]: I think something about segment anything model would be more in order to add here. Mask generation is a new machine learning task introduced with SAM

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+Generating masks that can to separate background and foreground (or other interesting properties) in an image can be done to facilitate learning, especially in semi- or unsupervised learning.
+
+### Human-in-the-loop
+
+For applications where humans are in the loop, masks highlight certain region of images for humans to validate. 
+
+## Task Variants
+
+### Binary Masks
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:17]: I think binary <-> non-binary masks are just representations in segmentation where you can switch between both instead of being mask generation variants

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+In binary masks, one region is labeled with ones (1) while another region is labeled with zeroes (0). 
+In this case, matrix multiplication easily retrieves the region of interest.
+
+### Non-Binary Masks
+
+In non-binary masks, the mask contains more than two classes. This is useful when more than one mask is generated for the same image since it makes dealing with the image easier. If there are non-overlapping labeled regions, it is straightforward to recover the mask by selecting which pixels in the image have the corresponding region. If they overlap, a unique way to recover a region might have to be put in place.
+
+### Segmentation Mask
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:26]: I think this is not a use-case but due to how mask generation is sort of another variant of segmentation and segmentation would be the bigger part IMO

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+Segmentation masks are the result of a segmentation algorithm. You can learn more about segmentation on its [task page](https://huggingface.co/tasks/image-segmentation).
+
+
+### Manually Labeled Mask
+
+Manually Labeled Mask are mask that have been generated, partially or entirely, by human annotations.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:33]: I think this is not really a use case of mask generation

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+Segmentation masks are the result of a segmentation algorithm. You can learn more about segmentation on its [task page](https://huggingface.co/tasks/image-segmentation).
+
+
+### Manually Labeled Mask
+
+Manually Labeled Mask are mask that have been generated, partially or entirely, by human annotations. 
+
+## Inference
+
+This section should have useful information about how to pull a model from Hugging Face Hub which is a part of a library specialized in a task and use it.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:37]: you can actually add how to use with transformers ⬇️ 
```python
from transformers import pipeline
generator =  pipeline("mask-generation", device = 0, points_per_batch = 256)
image_url = "https://huggingface.co/ybelkada/segment-anything/resolve/main/assets/car.png"
outputs = generator(image_url, points_per_batch = 256)
```

File: packages/tasks/src/tasks/mask-generation/about.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
+## Use Cases
+
+### Filtering an Image
+
+The mask might serve as an initial filter to eliminate irrelevant information. For instance, when monitoring vegetation in satellite imaging, a mask can be created to identify green spots, highlighting the relevant region of the image.
+
+### Pre-processing
+
+Generating masks that can to separate background and foreground (or other interesting properties) in an image can be done to facilitate learning, especially in semi- or unsupervised learning.
```
Comment (merveenoyan) [inline packages/tasks/src/tasks/mask-generation/about.md:9]: I think this might actually be used to train segmentation models

**Review by merveenoyan — COMMENTED** [review]
Thanks a lot for this PR <3

**Comment by bellabf** [comment]
> Thanks a lot for this PR <3

Hi, Merve! Thank you so much for the review. I had misunderstood the task variation part and I fixed it now following your suggestion. I tried to create a better naming for the second use case you mentioned, but I am a bit out of ideas.  In more plain terms, the idea is: 

* pre-processing= I know that the task only concern this part of the image, so I will use the mask to tell the model look here
* masked image modelling= I'll train the model to reconstruct the masked part of the image 

Thanks again <3

**Comment by merveenoyan** [comment]
@bellabf can you run the following in this repository?
```
sudo corepack enable
pnpm install

pnpm -r format
pnpm -r lint
```
and some files in this PR should change. Can you add, commit and push them?

**Comment by merveenoyan** [comment]
@bellabf @NebulaAnish if you could give me your HF handles I can add them at the end of the published page for credits

**Comment by bellabf** [comment]
> @bellabf @NebulaAnish if you could give me your HF handles I can add them at the end of the published page for credits

Hi, Merve! Thanks, it is [bisnotforbella](https://huggingface.co/bisnotforbella)

**Comment by NebulaAnish** [comment]
Hi, my handle is [nebulaanish](https://huggingface.co/nebulaanish). Thanks you.
